debugger/gdb-sgx-plugin/thread.py

```diff
@@ -0,0 +1,326 @@
+import gdb
+import math
+import tempfile
+from typing import List
+
+DEBUG = False
+
+INSTANCE_DEBUG_EXT = None
+
+class ThreadCloneBreakpoint(gdb.Breakpoint):
+    def __init__(self):
+        """
+        THIS BREAKPOINT IS DISABLED AS FUNCTION CALL MAY NOT BE RELIABLE
+
+        Set a breakpoint @ kernel/syscall.c::_SYS_myst_clone
+        Choosing this location because this is where the parent process/thread creates new process/thread
+        We can get the parent pid/tid from here
+        """
+        super(ThreadCloneBreakpoint, self).__init__('kernel/syscall.c:4094', internal=False)
```

Comment:
Assuming the breakpoint was supposed to be inside `_SYS_myst_clone`. Line number has already changed in `main`, 4094 is no longer within that function.
It would be better to avoid line number based breakpoints if possible. In this case `_SYS_myst_clone` seems like a wrapper around `myst_syscall_clone`, so you could possibly put a breakpoint on that.

Reply:
One technique we could use is to have function `myst_debug_hook_name(arg1, arg2)` and call them from within the enclave code. To allow setting internal breakpoints by name, also provide empty implementations of these functions on the host side (ensure that the host side empty implementations are not eliminated by the linker).

With such a setup, it ought to be possible to set breakpoints by name.

Reply:
Will create a hook function